src/GogeDao.sol

```diff
@@ -370,718 +476,955 @@ contract GogeDAO is Ownable {
+        return polls[_pollNum][_addr];
     }
 
-    function updateVetoEnabled(bool enabled) external onlyOwner() {
-        _updateVetoEnabled(enabled);
+    /// @notice A view method for returning a poll's unique metadata.
+    /// @param _pollNum unique poll identifier.
+    function getMetadata(uint256 _pollNum) public view returns (Metadata memory) {
+        return pollMap[_pollNum];
     }
 
-    function setTeamMember(address addr, bool value) internal {
-        if(value) {
-            (bool _isTeamMember, ) = isTeamMember(addr);
-            if(!_isTeamMember) teamMembers.push(addr);        
-        } else {
-            (bool _isTeamMember, uint8 s) = isTeamMember(addr);
-            if(_isTeamMember){
-                teamMembers[s] = teamMembers[teamMembers.length - 1];
-                teamMembers.pop();
-            } 
+    /// @notice A view method for returning a poll's current proportion of votes over circuating supply.
+    /// @param _pollNum unique poll identifier.
+    function getProportion(uint256 _pollNum) public view returns (uint256) {
+        return totalVotes[_pollNum] * 100 / IGogeERC20(governanceTokenAddr).getCirculatingMinusReserve();
+    }
+
+    /// @notice A view method for returning the current polls (by pollNum) are active, but expired.
+    /// @return expired array of pollNums that are active, but expired.
+    /// @return counter the amount of polls that are active, but expired.
+    function findExpiredPolls() public view returns (uint256[] memory expired, uint256 counter) {
+        uint256 l = activePolls.length;
+        expired = new uint256[](l);
+
+        for (uint256 i = 0; i < l; i++) {
+            uint256 endTime = pollEndTime[activePolls[i]];
+
+            if (block.timestamp >= endTime) {
+                expired[counter++] = activePolls[i];
+            }
         }
     }
 
-    function setVetoAuthority(address addr, bool value) internal {
-        require(vetoAuthority[addr] != value, "Already set");
-        vetoAuthority[addr] = value;
+    /// @notice A view method for returning whether a given poll is active.
+    /// @param  _pollNum unique poll identifier.
+    /// @return active whether poll is currently active.
+    function isActivePoll(uint256 _pollNum) public view returns (bool active) {
+        for (uint256 i = 0; i < activePolls.length; i++){
+            if (_pollNum == activePolls[i]) {
+                return true;
+            }
+        }
+        return false;
     }
 
-    function updateQuorum(uint256 amount) internal {
-        quorum = amount;
+    /// @notice A view method for returning the amount of active polls an author currently has.
+    /// @param  _account author's wallet address.
+    /// @return _num number of polls that are active, by author.
+    function getActivePollsFromAuthor(address _account) public view returns (uint256 _num) {
+        for (uint256 i = 0; i < activePolls.length; i++){
```

Comment:
Cache the length in a local variable outside of the for loop, use `++i`, and `uint256 i;`.